packages/databricks-vscode/src/configuration/ConnectionManager.ts

```diff
@@ -256,227 +239,81 @@ export class ConnectionManager {
         try {
-            if (typeof cluster === "string") {
-                cluster = await Cluster.fromClusterId(
-                    this._workspaceClient!.apiClient,
-                    cluster
-                );
-            }
-
-            if (
-                JSON.stringify(this.cluster?.details) ===
-                JSON.stringify(cluster.details)
-            ) {
-                return;
-            }
-
-            if (!skipWrite) {
-                this._projectConfigFile!.clusterId = cluster.id;
-                await this._projectConfigFile!.write();
-            }
-
-            if (cluster.state === "RUNNING") {
-                cluster
-                    .canExecute()
-                    .then(() => {
-                        this.onDidChangeClusterEmitter.fire(this.cluster);
-                    })
-                    .catch((e) => {
-                        NamedLogger.getOrCreate(Loggers.Extension).error(
-                            `Error while running code on cluster ${
-                                (cluster as Cluster).id
-                            }`,
-                            e
-                        );
-                    });
-            }
-
-            cluster
-                .hasExecutePerms(this.databricksWorkspace?.user)
-                .then(() => {
-                    this.onDidChangeClusterEmitter.fire(this.cluster);
-                })
-                .catch((e) => {
-                    NamedLogger.getOrCreate(Loggers.Extension).error(
-                        `Error while fetching permission for cluster ${
-                            (cluster as Cluster).id
-                        }`,
-                        e
-                    );
-                });
-
-            this.updateCluster(cluster);
-        } catch (e) {
+            this.login(config.authProvider);
+        } catch (e: any) {
             NamedLogger.getOrCreate("Extension").error(
-                "Attach Cluster error",
+                `Connection using "${config.authProvider.describe()}" failed`,
                 e
             );
-            window.showErrorMessage(
-                `Error in attaching cluster ${
-                    typeof cluster === "string" ? cluster : cluster.id
-                }`
+
+            await window.showWarningMessage(
+                `Connection using "${config.authProvider.describe()}" failed with error: "${
+                    e.message
+                }"."`
             );
-            await this.detachCluster();
         }
     }
 
-    async detachCluster(): Promise<void> {
-        if (!this.cluster && this._projectConfigFile?.clusterId === undefined) {
+    @onError({
+        popup: {prefix: "Can't attach cluster: "},
+        log: {logger: Loggers.Extension},
+    })
+    async attachCluster(clusterId: string): Promise<void> {
+        if (this.cluster?.id === clusterId) {
             return;
         }
+        this.configModel.set("clusterId", clusterId);
```

Comment:
Is `set` here returns a promise? We should return it or await, otherwise the error will be swallowed

Same for the detachCluster

Reply:
missed it.